src/txmempool.cpp

```diff
@@ -1055,11 +1055,36 @@ void CTxMemPool::TrimToSize(size_t sizelimit, std::vector<COutPoint>* pvNoSpends
     }
 }
 
-bool CTxMemPool::TransactionWithinChainLimit(const uint256& txid, size_t chainLimit) const {
+uint64_t CTxMemPool::CalculateDescendantMaximum(txiter entry) const {
+    // find parent with highest descendant count
+    std::vector<txiter> candidates;
```

Comment:
nit: more true to form to use a `std::stack` here, although I see they're not used otherwise in the codebase fwiw.

Reply:
nit,
```cpp
std::vector<txiter> candidates = {entry};
```
and remove `push_back` below.